code/modules/projectiles/guns/energy/laser.dm

```diff
@@ -197,12 +197,23 @@
 	can_knockdown = FALSE //Projectiles that pierce can not knockdown, no wall knockdowns.
 
 /obj/item/projectile/beam/laser/sniper/pierce
-	forcedodge = 1 // Can pierce one mob.
+	forcedodge = 1 // Can pierce one non wall thing.
 	speed = 0.5
+	/// Have we hit an r_wall? If we have, don't pierce it again so we don't become too effective on reinforced locations (AI sat)
+	var/hit_a_r_wall = FALSE
 
 /obj/item/projectile/beam/laser/sniper/pierce/prehit(atom/target)
-	if(isturf(target) && !forcedodge)
-		forcedodge++ //Increases force dodge before turf consumes it.
+	if(!isturf(target))
+		return ..() //Not a wall, act like a normal projectile
+	if(!istype(target, /turf/simulated/wall/r_wall))
+		if(!forcedodge) //We are a wall, do we need to increase pierce so we don't die in a wall?
+			forcedodge++ //Increases force dodge before turf consumes it.
+		return ..()
+	if(hit_a_r_wall) //We are a rwall, have we hit one before and should not pierce?
+		return ..() //We want it to pierce ONE reinforced wall, sorry, no easy 2 shotting the AI.
+	hit_a_r_wall = TRUE
+	if(!forcedodge) //We have not hit an rwall before, let us check if we need to increase
+		forcedodge++
 	..()
```

Comment:
Needs confirmation that logic works the same, but i think this is cleaner
```
if(istype(target, /turf/simulated/wall/r_wall))
    if(!hit_a_r_wall)
        hit_a_r_wall = TRUE;
        if(!forcedodge)
            forcedodge++
else if (isturf(target) || !forcedodge)
    forcedodge++
..()
```

Reply:
done!